### PR TITLE
Registry credentials configuration

### LINKED ISSUE #1574: add credentials to RegistryConfiguration
**What feature would you like to be added?**

Make it possible to add credentials to RegistryConfiguration.

During deployments we're hitting _rate limits_ of Docker Hub causing our Kubeone deployments to fail. We've tried patching containerd config.json with the [credentials of our registry](https://github.com/containerd/containerd/blob/main/docs/cri/registry.md#configure-registry-credentials) (harbor) instances, but those are [overwritten by KubeOne](https://github.com/kubermatic/kubeone/blob/master/pkg/scripts/render.go#L31).
As our registries over the internet internet, we're not allowed by the security department to make the Proxy Cache type _Public_.

This issue is related to #1551, #1470 and #1547 

**What are use cases of the feature?**

Use a _Private_ (not _Public_) registry Proxy Cache for caching Docker Hub artifacts. This allows for successful deployments that are not failing due to Docker Hub rate limits.

### PR DESCRIPTION
**Pull request by kron4eg**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1574

**Does this PR introduce a user-facing change?**:

```release-note
Registry credentials configuration
```

### CODE REVIEW

File: pkg/cmd/config.go [hunk 1]
```diff
@@ -610,6 +610,9 @@ containerRuntime:
   #     docker.io:
   #       mirrors:
   #       - http://plain-text2.tld
+  #       auth:
+  #         username: "uzer"
+  #         password: "myc00lp455w0rd"
```
Comment (xmudrii) [inline pkg/cmd/config.go:615]: Can we also document Auth and IdentityToken fields?

**Review by xmudrii — CHANGES_REQUESTED** [review]
Just a nit, but otherwise LGTM.

File: pkg/cmd/config.go (hunk 1 above)
Comment (kron4eg) [reply pkg/cmd/config.go]: The docs are in the generated API, this is just an example. But nether the less, done.

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve

**Pull request merged**